prow/plugins/size/gen_files_group.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package size
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/test-infra/prow/github"
+)
+
+const genConfigFile = ".generated_files"
+
+// ghFileClient scopes to the only relevant functionality we require of a github client.
+type ghFileClient interface {
+	GetFile(org, repo, filepath, commit string) ([]byte, error)
+}
+
+// GenFilesGroup is a logical collection of files. Check for a file's
+// inclusion in the group using the Match method.
+type GenFilesGroup struct {
+	Paths, FileNames, PathPrefixes, FilePrefixes map[string]bool
+}
+
+// NewGenFilesGroup reads the .generated_files file in the root of the repository
+// and any referenced path files (from "path-from-repo" commands).
+func NewGenFilesGroup(gc ghFileClient, owner, repo, sha string) (g *GenFilesGroup, err error) {
+	g = &GenFilesGroup{
+		Paths:        make(map[string]bool),
+		FileNames:    make(map[string]bool),
+		PathPrefixes: make(map[string]bool),
+		FilePrefixes: make(map[string]bool),
+	}
+
+	var bs []byte
+	bs, err = gc.GetFile(owner, repo, genConfigFile, sha)
+	if err != nil {
+		switch err.(type) {
+		case *github.FileNotFound:
+			return g, nil
+		default:
+			err = fmt.Errorf("GetFile error: %v", err)
+			return
```

Comment:
Likely a personal taste, but I think it's better to always use return arguments.